src/infrasys/base_quantity.py

```diff
@@ -79,6 +79,8 @@ def to_dict(self) -> dict[str, Any]:
         val = self.magnitude
         if isinstance(self.magnitude, np.ndarray):
             val = self.magnitude.tolist()
+        if isinstance(self.magnitude, np.generic):
```

Comment:
Would this ever conflict with the previous conditional? Should it be `elif` instead of `if`?

Reply:
Good call. I will make it elif